src/validator/keystore.ts

```diff
@@ -0,0 +1,33 @@
+/**
+ * @module validator
+ */
+
+import bls from "@chainsafe/bls-js";
+import fs from "fs";
+
+import {blsPrivateKeyToHex} from "../util/bytes";
+import {encryptKey, decryptKey} from "../util/encrypt";
+
+export default class Keystore {
+  private encryptedPrivateKey: string;
+  public publicKey: string;
+
+  public constructor(password: string) {
+    const keyPair = bls.generateKeyPair();
+
+    this.encryptedPrivateKey = encryptKey(blsPrivateKeyToHex(keyPair.privateKey), password);
+    this.publicKey = keyPair.publicKey.toHexString();
+  }
+
+  public privateKey(password: string): string {
+    return decryptKey(this.encryptedPrivateKey, password);
+  }
+
+  public saveKeys(outputFilePath: string): void {
+    try {
+      fs.writeFileSync(outputFilePath, JSON.stringify(this, null, 2));
```

Comment:
you should probably create missing directories

Reply:
ah, yes. An oversight. Will fix now